src/modules/swap/containers/SurplusModalSetup/index.tsx

```diff
@@ -16,12 +17,20 @@ export function SurplusModalSetup() {
 
   const { chainId } = useWalletInfo()
   const order = useOrder({ id: orderId, chainId })
+  const { showSurplus } = useGetSurplusData(order)
 
   const onDismiss = useCallback(() => {
     orderId && removeOrderId(orderId)
   }, [orderId, removeOrderId])
 
-  const isOpen = !!orderId
+  useEffect(() => {
+    // If we should NOT show the surplus, remove the orderId from the queue
+    if (showSurplus === false && orderId) {
+      removeOrderId(orderId)
+    }
+  }, [orderId, removeOrderId, showSurplus])
```

Comment:
Does it make sense to add to remove later? doesn't it make sense to not add in the queue in the first place?

Reply:
The problem is, by the time we are adding it, the FIAT estimation is not available and has to be fetched/calculated.

https://github.com/cowprotocol/cowswap/blob/a6455fd0fa33a48178314bdf9b760f5dd2796e99/src/legacy/state/orders/updaters/PendingOrdersUpdater.ts#L234

It's a whole other can of worms to have it at that stage

Reply:
okay, i though could be the case, NVM